orca-core/src/main/java/com/netflix/spinnaker/orca/pipeline/persistence/ExecutionRepository.java

```diff
@@ -156,6 +156,15 @@ Execution retrievePipelineForCorrelationId(@Nonnull String correlationId)
 
   List<String> retrieveAllExecutionIds(@Nonnull ExecutionType type);
 
+  @Nullable
+  default String getPartition() {
+    throw new UnsupportedOperationException();
+  }
+
+  default boolean handlesPartition(@Nullable String partition) {
```

Comment:
nit:
```suggestion
  default boolean handlesPartition(@Nullable String partitionOfExecution) {
```

otherwise it's confusing whose partition you're talking about, esp in the `if`